templates/common/vsphere/files/vsphere-hostname.yaml

```diff
@@ -0,0 +1,12 @@
+filesystem: "root"
+mode: 0755
+path: "/usr/local/bin/vsphere-hostname.sh"
+contents:
+  inline: |
+    #!/usr/bin/env bash
+    set -e
+
+    if hostname=$(/bin/vmtoolsd --cmd 'info-get guestinfo.hostname'); then
```

Comment:
is this binary present in the RHCOS system today? should we inform RHCOS people that please don't remove this binary in later versions as we depend on it.

Reply:
Yeah its in RHCOS today.  I tested out all the commands prior to scripting and using systemd to execute.

Reply:
It's not in FCOS though and we do want to eventually containerize it (once vmware stops fighting that).

Reply:
@cgwalters I have an example of running open-vm-tools in a container on fcos.  Using @dav1x blog post as an example: https://developers.redhat.com/blog/2017/03/23/containerizing-open-vm-tools-part-1-the-dockerfile-and-constructing-a-systemd-unit-file/

It would be interesting if I could query guestinfo by using an `exec` but I have yet to try.

Reply:
https://gist.github.com/jcpowermac/23ea90e549c02b43758662f90d83a9f1

Reply:
iirc we layer `vmtoolsd` into OKD's machine-os-content, so it'll be on master/worker nodes after bootstrapping the cluster.
Is script also present on bootstrap hosts?

Reply:
@LorbusChris nope I didn't add it to bootstrap.  bootstrap is fine as `localhost`.

Reply:
I had to do something similar to this when I converted the container to runc. The exec should work fine:

I changed the systemd unit file to call this shell script instead. 
```
#!/bin/sh
if [ -f /tmp/open-vm-tools-bash-env ]
  then
    source /tmp/open-vm-tools-bash-env
fi

COMMAND=/usr/local/bin/vmware-toolbox-cmd
if [ ! -e $COMMAND ]
  then
    echo 'runc exec -t open-vm-tools vmware-toolbox-cmd "$@"' > /usr/local/bin/vmware-toolbox-cmd
    chmod +x /usr/local/bin/vmware-toolbox-cmd
fi
exec /usr/bin/VGAuthService -s &
exec /usr/bin/vmtoolsd
```